Zend/zend_API.c

```diff
@@ -1983,30 +1983,42 @@ ZEND_API void zend_check_magic_method_implementation(const zend_class_entry *ce,
 			zend_error(error_type, "Method %s::%s() must take exactly 1 argument", ZSTR_VAL(ce->name), ZEND_GET_FUNC_NAME);
 		} else if (QUICK_ARG_SHOULD_BE_SENT_BY_REF(fptr, 1)) {
 			zend_error(error_type, "Method %s::%s() cannot take arguments by reference", ZSTR_VAL(ce->name), ZEND_GET_FUNC_NAME);
+		} else if (ZEND_TYPE_IS_SET(fptr->common.arg_info[0].type) && ZEND_TYPE_FULL_MASK(fptr->common.arg_info[0].type) != MAY_BE_STRING) {
+			zend_error(error_type, "Type of first argument of %s::%s() must be string", ZSTR_VAL(ce->name), ZEND_GET_FUNC_NAME);
```

Comment:
TODO: Improve these error messages following #5092.

Reply:
Actually, I wasn't right in my previous comment in the other PR: as these error messages are related to invalid signatures (rather than argument passing), so I think a message adopted from regular inheritance checks would be a better fit: `Declaration of %s::%s(...) must be compatible with %s::%s(string $name)`. And this message could replace all the other error messages (`Method %s::%s() must take exactly 1 argument` and `Method %s::%s() cannot take arguments by reference`).

And this raises a question in me: shouldn't we allow all compatible types rather than just no type and `string`? What I mean: the current implementation only accepts `Foo::__get(string $name)` and `Foo::__get($name)`, while `Foo::__get(int|string $name)` is in theory also compatible. Or it is not a sane idea?

Reply:
@kocsismate The `__get` method access the properties of a class. Can classes have properties with numbers only? If so, then we can allow `int`.

Reply:
>  as these error messages are related to invalid signatures (rather than argument passing), so I think a message adopted from regular inheritance checks would be a better fit: Declaration of %s::%s(...) must be compatible with %s::%s(string $name). And this message could replace all the other error messages (Method %s::%s() must take exactly 1 argument and Method %s::%s() cannot take arguments by reference).

That's not a bad idea actually. I'll work on that :)

Reply:
> Can classes have properties with numbers only

No, of course not :) I was just wondering if following the regular inheritance rules would be a good idea or not. I think it would increase consistency, but surely, it's not very practical to define `int` as a type of the `$name` practical. :)

Reply:
> Actually, I wasn't right in my previous comment in the other PR: as these error messages are related to invalid signatures (rather than argument passing), so I think a message adopted from regular inheritance checks would be a better fit: `Declaration of %s::%s(...) must be compatible with %s::%s(string $name)`. And this message could replace all the other error messages (`Method %s::%s() must take exactly 1 argument` and `Method %s::%s() cannot take arguments by reference`).

This is not possible, not for all methods. Magic methods usually don't come from a parent class.

Reply:
Well, classes *can* have integer properties; not sure if these could be private/protected, though.

Reply:
> Magic methods usually don't come from a parent class.

That's true. I was also a bit hesitant first, but it might not be that bad an idea for example to write: `Declaration of Foo::__get(array $name) must be compatible with Foo::__get(string $name)`. I admit though that it's a little bit weird to have the same class name twice.

Nevertheless, the error message itself is an implementation detail, so in my opinion you shouldn't work on it too much until the RFC gets accepted. :)

Reply:
> Well, classes can have integer properties; not sure if these could be private/protected, though.

How https://3v4l.org/t7OTC?

Reply:
I am referring to dynamic properties, e.g. https://3v4l.org/sVHSt.

Reply:
But these are converted to string, so no problem here. Sorry for the noise.

Reply:
Thanks for checking that! 

As said in the chat, I'll adapt the RFC with Union Types. Don't want to block or remove any existent functionality, just endure that typed are use correctly :)